### PR TITLE
SEARCH-2589: Check Solr6 subsystem doesn't include the ShardRegistry …

### PR DESCRIPTION
**Pull request by aborroy**
…reference for Community

### CODE REVIEW

File: repository/src/test/java/org/alfresco/repo/search/impl/solr/SolrSubsystemTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    private SOLRTrackingComponentImpl solr6TrackingBean;
+
+    @Before
+    public void buildElasticsearchClient() throws Exception
+    {
+        SwitchableApplicationContextFactory subsystemManager = ((SwitchableApplicationContextFactory) applicationContext
+                    .getBean("Search"));
+        ApplicationContext solrContext = subsystemManager.getApplicationContext();
+        solr6TrackingBean = (SOLRTrackingComponentImpl) solrContext.getBean("search.trackingComponent");
```
Comment (killerboot) [inline repository/src/test/java/org/alfresco/repo/search/impl/solr/SolrSubsystemTest.java:47]: Please add a check for this bean to be not null.

File: repository/src/test/java/org/alfresco/repo/search/impl/solr/SolrSubsystemTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
```
Comment (killerboot) [inline repository/src/test/java/org/alfresco/repo/search/impl/solr/SolrSubsystemTest.java:5]: please change to 2020

File: repository/src/test/java/org/alfresco/repo/search/impl/solr/SolrSubsystemTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+package org.alfresco.repo.search.impl.solr;
+
+import org.alfresco.repo.management.subsystems.SwitchableApplicationContextFactory;
+import org.alfresco.repo.solr.SOLRTrackingComponentImpl;
+import org.alfresco.util.BaseSpringTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.context.ApplicationContext;
+
+public class SolrSubsystemTest extends BaseSpringTest
```
Comment (killerboot) [inline repository/src/test/java/org/alfresco/repo/search/impl/solr/SolrSubsystemTest.java:36]: Is "solr6" subsystem enabled in this case?
The default is "noindex".

Comment (aborroy) [reply repository/src/test/java/org/alfresco/repo/search/impl/solr/SolrSubsystemTest.java:36]: Travis is starting the test with `-Dindex.subsystem.name=solr6`
https://github.com/Alfresco/alfresco-community-repo/blob/master/.travis.yml#L131

**Pull request merged**